### DISCUSSION TITLE
Suggestion - add progress bar with current download speed

### DISCUSSION THREAD

**Discussion by Andrej730** [houtianze/bypy#637] [Ideas]
My suggestion is to add ASCII progress bar with current download speed or just some interfomation to see that's connection happened and download is in progress. 
Currently with large files you just wait in hope everything worked fine but you can't be sure if speed is okay to finish downloading in reasonable time.

Update: progress bar appears but only after a few minutes, I guess it would be nice to have more verbose way of connection then.

**Comment by houtianze** [comment]
Thanks for the suggestion, however, I think the ascii progress is already there if I'm not wrong.

**Reply by Andrej730** [reply]
yeah, the problem was that it was taking a while to connect and in the meantime it's hard to guess is it working or not

**Reply by houtianze** [reply]
The progress par doesn't "connect" anything, and has speed and ETA info, and I don't understand why you have to "guess" it.

**Reply by Andrej730** [reply]
I meant it was taking awhile (like 5 mins) to connect to pan baidu (so it was before progress bar showing up) and it wasn't clear at what stage is the connection.

**Comment by houtianze** [comment]
The delay is related to speed, iirc, the progress will be calculated after
1st chunk (20Mb iinw) is uploaded/downloaded, that's when it gets data of
the speed to display the progress bar, but as the speed is so slow now, it
takes some time for the first paint of the progress bar.

**Reply by Andrej730** [reply]
So basically if I'll set `dl_chunk_size` initializing `ByPy` then the information will appear faster?
Is there any downsides to reducing `dl_chunk_size`?

**Comment by houtianze** [comment]
Yes, I think it should appear faster. I’m not aware of any downside, but I almost never tried that. I think there is a program argument like —chunk or similar to set it.On 3 Feb 2023, at 10:19 PM, Andrej730 ***@***.***> wrote:﻿
So basically if I'll set dl_chunk_size initializing ByPy then the information will appear faster?
Is there any downsides to reducing dl_chunk_size?
